test/suites/integration/byoc_test.go

```diff
@@ -292,7 +292,11 @@ func findHostedZone(client *route53.Route53, name string) (*route53.HostedZone,
 	if len(out.HostedZones) == 0 {
 		return nil, nil
 	}
-	return out.HostedZones[0], nil
+	hz := out.HostedZones[0]
+	if aws.StringValue(hz.Name) != name {
+		return nil, nil
+	}
```

Comment:
why name should not be equal? list zones should apply dns name filter
`out, err := client.ListHostedZonesByName(&route53.ListHostedZonesByNameInput{DNSName: &name})`

Reply:
> "Fix bug where e2e test will pick up existing other hosted zone and delete it"

There was intention to use existing hosted zone if test was terminated and zone leaked. Test re-run should pick-up previous zone and cleanup after. Otherwise need to manually delete them from route53.

Reply:
ListHostedZonesByName always returns all hosted zones in the account. DNSName field only guarantees the first element that is getting picked up, and you still get other hosted zones in the following elements.